src/GraphQL.Conventions/Adapters/GraphTypeAdapter.cs

```diff
@@ -29,45 +29,28 @@ public ISchema DeriveSchema(GraphSchemaInfo schemaInfo)
             var interfaces = types
                 .Where(t => t.IsInterfaceType && !t.IsInputType && t.PossibleTypes.Any())
                 .GroupBy(t => t.Name)
-                .Select(g => g.First());                
-            var myPossibleTypes = interfaces
+                .Select(g => g.First());
+            var possibleTypes = interfaces
                 .Where(t => !t.IsIgnored)
                 .SelectMany(t => t.PossibleTypes)
-                .Select(x => OmitNonRootTypes(x))
+                .Select(typeInfo => (typeInfo.IsIgnored || !typeInfo.IsNullable || typeInfo.Interfaces.Any(x => x.IsIgnored == true))
```

Comment:
Maybe we can rewrite this like so:
```
.Where(typeInfo => !typeInfo.IsIgnored 
   || typeInfo.IsNullable 
   || typeInfo.Interfaces.All(x => !x.IsIgnored)
)
```
I think that way we can avoid the ternary/null and the `Where(x => x != null)`-operations here. 😃

Reply:
this is a result of merging with the latest Conventions.NET master branch, 
so I would avoid any changes here, I don't want to drag more things to support and maintain